src/App.tsx

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import RestaurantList from './components/RestaurantList';
+import mockData from "./data/mockData.json";
+import { Category, Restaurant, Sort } from './types/Restaurant';
+import SelectBox from './components/SelectBox';
+import Header from './components/Header';
+import styled from 'styled-components';
+
+interface AppState {
+  restaurants: Restaurant[];
+  sortBy: Sort;
+  categorizeBy: Category;
+}
+class App extends React.Component<{}, AppState> {
+
+  constructor(props: Readonly<{}> | {}) {
+    super(props);
+    this.state = { restaurants: mockData, sortBy: 'name', categorizeBy: 'all' };
+  }
+
+  updateSortBy(e: React.ChangeEvent<HTMLSelectElement>) {
+    this.setState({ sortBy: e.target.value as Sort });
+  }
+
+  updateCategorizeBy(e: React.ChangeEvent<HTMLSelectElement>) {
+    this.setState({ categorizeBy: e.target.value as Category });
+  }
+
+  render() {
+    const { sortBy, categorizeBy, restaurants } = this.state;
+
+    return (
+      <>
+        <Header />
+        <SelectContainer>
+          <SelectBox
+            setState={(e) => this.updateCategorizeBy(e)}
```

Comment:
1. setState 라는 props 명은 오해의 여지가 있어요. onSelect 등 유저 액션을 표현하는 함수명은 어떨까요?
2. 이렇게 render 하위에서 선언된 화살표 함수는 SelectBox 컴포넌트가 재렌더링을 할 여지가 있습니다. 파라미터가 이벤트 타입만 들어간다면 화살표 함수를 생략해주는 것이 좋겠네요!

```suggestion
onSelect={this.updateCategorizeBy}
```